components/cdc/src/endpoint.rs

```diff
@@ -236,15 +247,23 @@ impl<T: 'static + RaftStoreRouter<RocksEngine>> Endpoint<T> {
         raft_router: T,
         observer: CdcObserver,
         store_meta: Arc<Mutex<StoreMeta>>,
+        txn_extra_receiver: mpsc::Receiver<TxnExtra>,
     ) -> Endpoint<T> {
         let workers = Builder::new().name_prefix("cdcwkr").pool_size(4).build();
-        let tso_worker = Builder::new().name_prefix("tso").pool_size(1).build();
+        let misc_worker = Builder::new().name_prefix("tso").pool_size(1).build();
+        let scheduler_clone = scheduler.clone();
+        misc_worker.spawn(txn_extra_receiver.for_each(move |txn_extra| {
+            if let Err(e) = scheduler_clone.schedule(Task::TxnExtra(txn_extra)) {
```

Comment:
Why create another thread just to transfer the task to another sender?
You can define a scheduler trait in TiKv just as following:
```
pub trait TxnExtraScheduler {
   fn schedule(&self, txn_extra: TxnExtra);
}
```
And then implement it in CDC so that you can pass a Box<dyn TxnExtraScheduler> to `RaftKV`.